--- browser/Makefile.am.orig	2004-08-13 10:58:36.625931416 +0200
+++ browser/Makefile.am	2004-08-13 10:58:47.772236920 +0200
@@ -125,7 +125,7 @@
 	scp tables.sql server.cs server.asmx monodoc.dll root@www.go-mono.com:
 
 install-data-local:
-	$(GACUTIL) /i monodoc.dll /f /package gtk-sharp /gacdir $(DESTDIR)$(libdir)
+	$(GACUTIL) /i monodoc.dll /f /package gtk-sharp /root $(DESTDIR)$(libdir)
 
 uninstall-local:
 	$(GACUTIL) /u monodoc /package gtk-sharp /gacdir $(DESTDIR)$(libdir)
