diff -aurN monodoc-1.0.6.orig/browser/Makefile.am monodoc-1.0.6/browser/Makefile.am
--- monodoc-1.0.6.orig/browser/Makefile.am	2005-04-21 19:58:27.000000000 +0200
+++ monodoc-1.0.6/browser/Makefile.am	2005-04-21 20:02:39.000000000 +0200
@@ -60,7 +60,7 @@
 cs2ecma_sources = \
 	$(srcdir)/cs2ecma.cs
 
-browser_assemblies = -pkg:gtkhtml-sharp -pkg:glade-sharp -r:System.Web.Services
+browser_assemblies = -pkg:gtkhtml-sharp-2.0 -pkg:glade-sharp-2.0 -r:System.Web.Services
 
 EXTRA_DIST = \
 	$(monodoc_sources) $(assembler_sources) \
@@ -150,7 +150,7 @@
 	scp tables.sql server.cs server.asmx monodoc.dll root@www.go-mono.com:
 
 install-data-local:
-	$(GACUTIL) /i monodoc.dll /f /package gtk-sharp /gacdir $(GACDIR) /root $(GACROOT)
+	$(GACUTIL) /i monodoc.dll /f /package gtk-sharp-2.0 /gacdir $(GACDIR) /root $(GACROOT)
 
 uninstall-local:
-	$(GACUTIL) /u monodoc /package gtk-sharp /gacdir $(GACDIR)
+	$(GACUTIL) /u monodoc /package gtk-sharp-2.0 /gacdir $(GACDIR)
diff -aurN monodoc-1.0.6.orig/configure.in monodoc-1.0.6/configure.in
--- monodoc-1.0.6.orig/configure.in	2005-04-21 19:58:27.000000000 +0200
+++ monodoc-1.0.6/configure.in	2005-04-21 19:59:22.000000000 +0200
@@ -41,7 +41,7 @@
 if test "x$with_gtk" = "xyes" ; then
   GTKSHARP_REQUIRED_VERSION=0.91
   MONO_REQUIRED_VERSION=0.90
-  PKG_CHECK_MODULES(BASE_DEPENDENCIES, gtk-sharp >= $GTKSHARP_REQUIRED_VERSION gtkhtml-sharp >= $GTKSHARP_REQUIRED_VERSION glade-sharp >= $GTKSHARP_REQUIRED_VERSION)
+  PKG_CHECK_MODULES(BASE_DEPENDENCIES, gtk-sharp-2.0 >= $GTKSHARP_REQUIRED_VERSION gtkhtml-sharp-2.0 >= $GTKSHARP_REQUIRED_VERSION glade-sharp-2.0 >= $GTKSHARP_REQUIRED_VERSION)
 	PKG_CHECK_MODULES(BASE_DEPENDENCIES_RUNTIME, mono >= $MONO_REQUIRED_VERSION, has_mono=true, has_mono=false)
 	PKG_CHECK_MODULES(BASE_DEPENDENCIES_RUNTIME, mint >= $MONO_REQUIRED_VERSION, has_mint=true, has_mint=false)
 	if test "x$has_mono" = "xtrue"; then
--- monodoc-1.0.6/monodoc.pc.in~	2005-05-24 21:38:01.906304160 +0200
+++ monodoc-1.0.6/monodoc.pc.in	2005-05-24 21:38:21.599310368 +0200
@@ -1,8 +1,8 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@libdir@
+libdir=/usr/lib
 
 Name: Monodoc
 Description: Monodoc - Mono Documentation Browser
 Version: @VERSION@
-Libs: /r:${libdir}/mono/gtk-sharp/monodoc.dll 
+Libs: /r:${libdir}/mono/gtk-sharp-2.0/monodoc.dll 
